File: MyLearning/DsaPractice/Arrays/CreateAllSubArrays.cs
namespace DsaPractice.Arrays
{
    internal class CreateAllSubArrays
    {
        public void CreateAllSubArrayss()
        {
            int[] input = { 1, 2, 3, 4, 5 };
            List<List<int>> subArrays = new List<List<int>>();

            for (int i = 0; i < input.Length; i++)
            {
                List<int> current = new List<int>();

                for (int j = i; j < input.Length; j++)
                {
                    current.Add(input[j]);
                    subArrays.Add(new List<int>(current));
                }
            }

            foreach (var subArray in subArrays)
            {
                Console.WriteLine(string.Join(", ", subArray));
            }
        }
    }
}

File: MyLearning/PythonPractice/dsa_practice/arrays/create_all_sub_arrays.py
"""Create all contiguous subarrays."""

# C# CreateAllSubArrays.CreateAllSubArrayss

def create_all_sub_arrayss(input_list=None):
    """Return all contiguous subarrays.

    Problem: Generate all contiguous subarrays of an input list.
    Approach: Expand a window from each start index and capture each subarray.
    Time complexity: O(n^2)
    Space complexity: O(n^2)
    """
    if input_list is None:
        input_list = [1, 2, 3, 4, 5]

    sub_arrays = []

    for i in range(len(input_list)):
        current = []
        for j in range(i, len(input_list)):
            current.append(input_list[j])
            sub_arrays.append(list(current))

    return sub_arrays
